source/winAPI/sessionTracking.py

```diff
@@ -106,110 +90,105 @@ class WindowsTrackedSession(enum.IntEnum):
+			isWindowsLocked = False
+		else:
+			try:
+				isWindowsLocked = _isWindowsLocked_checkViaSessionQuery()
+			except Exception as e:
+				self._recordLockStateTrackingFailure(e)  # Report error repeatedly, attention is required.
+				##
+				# For security it would be best to treat unknown as locked.
+				# However, this would make NVDA unusable.
+				# Instead, the user should be warned via UI, and allowed to mitigate the impact themselves.
+				# See usage of L{warnSessionLockStateUnknown}.
+				isWindowsLocked = False
+
+		if self._prev_isWindowsLockedValue != isWindowsLocked:
+			from utils.security import postSessionLockStateChanged
+			postSessionLockStateChanged.notify(isNowLocked=isWindowsLocked)
 
+		# set prev value for comparison during the next cache cycle
+		self._prev_isWindowsLockedValue = isWindowsLocked
+		return isWindowsLocked
 
-def _hasLockStateBeenTracked() -> bool:
-	"""
-	Checks if NVDA is aware of a session lock state change since NVDA started.
-	"""
-	return bool(_currentSessionStates.intersection({
-		WindowsTrackedSession.SESSION_LOCK,
-		WindowsTrackedSession.SESSION_UNLOCK
-	}))
+	def _recordLockStateTrackingFailure(self, error: Optional[Exception] = None):
+		log.error(
+			"Unknown lock state, unexpected, potential security issue, please report.",
+			exc_info=error
+		)  # Report error repeatedly, attention is required.
+		##
+		# For security it would be best to treat unknown as locked.
+		# However, this would make NVDA unusable.
+		# Instead, the user should be warned, and allowed to mitigate the impact themselves.
+		# See usage of L{warnSessionLockStateUnknown}.
+		self._windowsLockStateIsUnknown = True
 
 
-def _recordLockStateTrackingFailure(error: Optional[Exception] = None):
-	log.error(
-		"Unknown lock state, unexpected, potential security issue, please report.",
-		exc_info=error
-	)  # Report error repeatedly, attention is required.
-	##
-	# For security it would be best to treat unknown as locked.
-	# However, this would make NVDA unusable.
-	# Instead, the user should be warned, and allowed to mitigate the impact themselves.
-	# Reporting is achieved via _sessionQueryLockStateHasBeenUnknown exposed with
-	# L{hasLockStateBeenUnknown}.
-	global _sessionQueryLockStateHasBeenUnknown
-	_sessionQueryLockStateHasBeenUnknown = True
+def initialize() -> None:
+	"""
+	This function must be called late during NVDA initialization.
+
+	During NVDA initialization,
+	core._initializeObjectCaches needs to cache the desktop object,
+	regardless of lock state.
+	Security checks may cause the desktop object to not be set if NVDA starts on the lock screen.
+	As such, during initialization, NVDA should behave as if Windows is unlocked.
+	sessionTracking.initialize is called when NVDA initialization is complete, which sets _currentSessionState
+	"""
+	global _currentSessionState
+	log.debug("initializing session tracking")
+	_currentSessionState = _CurrentSessionState()
 
 
 def isWindowsLocked() -> bool:

```

Comment:
Let's deprecate this and fix the name on master:

```suggestion
def isWindowsLocked() -> bool:
	return isWindowsLockScreenActive()

def isWindowsLockScreenActive() -> bool:
	...
```